packages/frontend/app/routes/tenants.$tenantId.tsx

```diff
@@ -0,0 +1,315 @@
+  }
+
+  return json({ tenant })
+}
+
+export default function ViewTenantPage() {
+  const { tenant } = useLoaderData<typeof loader>()
+  const response = useActionData<typeof action>()
+  const navigation = useNavigation()
+  const formAction = useFormAction()
+  const submit = useSubmit()
+  const dialogRef = useRef<ConfirmationDialogRef>(null)
+
+  const isSubmitting = navigation.state === 'submitting'
+  const currentPageAction = isSubmitting && navigation.formAction === formAction
+
+  const [formData, setFormData] = useState<FormData>()
+
+  const submitHandler = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault()
+    setFormData(new FormData(event.currentTarget))
+    dialogRef.current?.display()
+  }
+
+  const onConfirm = () => {
+    if (formData) {
+      submit(formData, { method: 'post' })
+    }
+  }
+
+  return (
+    <div className='pt-4 flex flex-col space-y-4'>
+      <div className='flex flex-col rounded-md bg-offwhite px-6'>
+        <PageHeader className='!justify-end'>
+          <Button aria-label='go back to tenants page' to='/tenants'>
+            Go to tenants page
+          </Button>
+        </PageHeader>
+        <div className='grid grid-cols-1 py-3 gap-6 md:grid-cols-3 border-b border-pearl'>
+          <div className='col-span-1 pt-3'>
+            <h3 className='text-lg font-medium'>General Information</h3>
+            <p className='text-sm'>
+              Created at {new Date(tenant.createdAt).toLocaleString()}
+            </p>
+            <ErrorPanel errors={response?.errors.general.message} />
+          </div>
+          <div className='md:col-span-2 bg-white rounded-md shadow-md'>
+            <Form method='post' replace preventScrollReset>
+              <fieldset disabled={currentPageAction}>
+                <div className='w-full p-4 space-y-3'>
+                  <Input type='hidden' name='id' value={tenant.id} />
+                  <Input
+                    label='Tenant ID'
+                    value={tenant.id}
+                    disabled
+                    readOnly
+                  />
+                  <Input
+                    label='Public Name'
+                    defaultValue={tenant.publicName ?? undefined}
+                    error={response?.errors.general.fieldErrors.publicName}
+                  />
+                  <Input
+                    label='Email'
+                    defaultValue={tenant.email ?? undefined}
+                    error={response?.errors.general.fieldErrors.email}
+                  />
+                </div>
+                <div className='flex justify-end p-4'>
+                  <Button
+                    aria-label='save general information'
+                    type='submit'
+                    name='intent'
+                    value='general'
+                  >
+                    {currentPageAction ? 'Saving ...' : 'Save'}
+                  </Button>
```

Comment:
This, nor any of the fields, arent working for me. Doesnt save.

I started the frontend and accessed as operator. I went into the tenant record and entered a name, saved, refreshed, and it was gone. I went to the tenant page and still saw no name or email. Same for all the other fields.

I see the `general` and `ip` cases are empty in the action

Reply:
@BlairCurrey yeah,  I pushed the missing code to have the input "name" attributes.